app.py

```diff
@@ -17,45 +19,73 @@
 	password = db.Column(db.Text, nullable=False)
 	is_admin = db.Column(db.Boolean, nullable=False)
 	is_banned = db.Column(db.Boolean, nullable=False)
+	is_typing = db.Column(db.Boolean, nullable=False)
 
 
 class Subjects(db.Model):
-	id = db.Column(db.BigInteger, primary_key=True)
+	id = db.Column(db.Integer, primary_key=True)
 	subject = db.Column(db.Text, unique=True, nullable=False)
 
 
 class Groups(db.Model):
-	id = db.Column(db.BigInteger, primary_key=True)
+	id = db.Column(db.Integer, primary_key=True)
 	groups = db.Column(db.Text, unique=True, nullable=False)
 
 
 class Posts(db.Model):
-	id = db.Column(db.BigInteger, primary_key=True)
-	user = db.Column(db.BigInteger, db.ForeignKey('users.id'), nullable=False)
-	subject = db.Column(db.BigInteger, db.ForeignKey('subjects.id'), nullable=False)
+	id = db.Column(db.Integer, primary_key=True)
+	user = db.Column(db.Integer, db.ForeignKey('users.id'), nullable=False)
+	subject = db.Column(db.Integer, db.ForeignKey('subjects.id'), nullable=False)
 	body = db.Column(db.Text, nullable=False)
-	group = db.Column(db.BigInteger, db.ForeignKey('groups.id'))
+	group = db.Column(db.Integer, db.ForeignKey('groups.id'))
 	time = db.Column(db.DateTime, nullable=False, default=datetime.now())
 
 
-@app.route('/signup', methods=['POST'])
+
+@app.route("/signup", methods=['POST'])
 def signup():
-	...
+	password: str = request.form['password']
+	username: str = request.form['username']
+	if not username or not password:
+		return make_response(jsonify({'task': 'login', 'status': 'failed', 'reason': 'Username and password are required'}), 400)
+	if Users.query.filter_by(username=username).all():
+		return make_response(jsonify({'task': 'signup', 'status': 'failed', 'reason': 'username already exists'}), 409)
+	if len(password) < 10:
+		return make_response(jsonify({'task': 'login', 'status': 'failed', 'reason': 'password must be more then 10 characters'}), 400)
+	hashed_password = generate_password_hash(password, method='sha256')
+	new_user = Users(username=username, password=hashed_password, is_admin=False, is_banned=False, is_typing=False)
+	db.session.add(new_user)  # adding the new user to the db
+	db.session.commit()
+	return make_response(jsonify({'task': 'signup', 'status': 'success'}), 200)
 
 
 @app.route('/login', methods=['POST'])
 def login():
-	...
+	password: str = request.form['password']
+	username: str = request.form['username']
+	if not username or not password:
+		return make_response(jsonify({'task': 'login', 'status': 'failed', 'reason': 'Username and password are required'}), 400)
+	if len(password) < 10:
+		return make_response(jsonify({'task': 'login', 'status': 'failed', 'reason': 'password must be more then 10 characters'}), 400)
+	# checking if the user and password are in the db
+	user_ls: list[Users] = Users.query.filter_by(username=username).all()
+	if user_ls and check_password_hash(user_ls[0].password, password):
+		session['username'] = True
```

Comment:
I ment add username key with the actual username value, maybe even add is_admin to the session